### PR TITLE
Add annotations to services defined in Charts 

### PR DESCRIPTION
**Pull request by etiennetremel**
- add annotations to services defined in Charts
- ~make use service type from values.yaml for tracing services~

### CODE REVIEW

**Comment by costinm** [comment]
Jaeger is getting reverted - so you'll likely get some conflicts.

Let's move it to master if possible, I don't think it's release blocking.

**Comment by etiennetremel** [comment]
That's unfortunate, I rebased the branch. This is definitely not a blocker but nice to have. Currently I have a workaround to patch some of the services because of the missing annotations field.

**Comment by etiennetremel** [comment]
Fixed conflicts since Jaeger is back again (https://github.com/istio/istio/pull/5840)

File: install/kubernetes/helm/istio/charts/tracing/templates/service-jaeger.yaml [hunk 1]
```diff
@@ -22,7 +26,7 @@ items:
         targetPort: {{ .Values.service.uiPort }}
     selector:
       app: jaeger
-    type: LoadBalancer
+    type: {{ .Values.service.type }}
```
Comment (objectiser) [inline install/kubernetes/helm/istio/charts/tracing/templates/service-jaeger.yaml]: This needs to be defined as a different type to the collector service - so can you create a different property in the values.yaml (e.g. .Values.service.uiType) which should be set to `LoadBalancer`?

File: install/kubernetes/helm/istio/charts/tracing/templates/service.yaml [hunk 2]
```diff
@@ -38,5 +42,5 @@ items:
         targetPort: {{ .Values.service.uiPort }}
     selector:
       app: jaeger
-    type: LoadBalancer
+    type: {{ .Values.service.type }}
```
Comment (objectiser) [inline install/kubernetes/helm/istio/charts/tracing/templates/service.yaml]: Same as above, should use a different property, e.g. `.Values.service.uiType` set to `LoadBalancer`.

File: install/kubernetes/helm/istio/charts/tracing/templates/service-jaeger.yaml (hunk 1 above)
Comment (etiennetremel) [reply install/kubernetes/helm/istio/charts/tracing/templates/service-jaeger.yaml]: in fact, much better. I just made the change. Thanks

File: install/kubernetes/helm/istio/charts/tracing/templates/service.yaml [hunk 3]
```diff
@@ -38,4 +42,5 @@ items:
         targetPort: {{ .Values.service.uiPort }}
     selector:
       app: jaeger
+    type: {{ .Values.service.uiType }}
```
Comment (gyliu513) [inline install/kubernetes/helm/istio/charts/tracing/templates/service.yaml]: We do not want to expose service type here, and intend to use Gateway to expose those services. 

Suggest your PR focus on adding `annotations` but do not touch service type in this PR

Comment (etiennetremel) [reply install/kubernetes/helm/istio/charts/tracing/templates/service.yaml]: Just removed any changes that are unrelated to the annotations.

**Pull request merged**